### ISSUE TITLE
Too long error message for recursive generic types

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#26289]
```dart
@optionalTypeArgs
abstract class Entity<TParent extends Entity> {}
class EntityMetadata<T extends Entity>{}
class EntityType {
  final EntityMetadata<Entity> entityMetadata;
//                     ^^^^^^
}
```
![image](https://cloud.githubusercontent.com/assets/405837/14610619/fd46276e-058e-11e6-91a5-6b90479d605f.png)

strong mode enabled

> Dart VM version: 1.16.0-edge.bab552edaf1080fdca423029128516e1c81a3279 (Fri Apr 15 13:25:32 2016) on "linux_x64"

**Comment by jmesserly** [comment]
related to https://github.com/dart-lang/sdk/issues/25530 ?

**Comment by zoechi** [comment]
seems to be fixed (dup of #26970)
